appservice/src/deploy/getDeployNode.ts

```diff
@@ -10,6 +10,10 @@ import { localize } from '../localize';
 import { getWorkspaceSetting } from '../utils/settings';
 import { AppSource, IDeployContext } from './IDeployContext';
 
+function isAzExtTreeItem(ti: unknown): ti is AzExtTreeItem {
+    return (ti as AzExtTreeItem).fullId !== undefined;
```

Comment:
It's probably not really necessary in practice, but in principle this might be slightly more resilient:
```suggestion
    return (ti as AzExtTreeItem).fullId !== undefined && (ti as AzExtTreeItem).fullId !== null;
```